sphinx/language_guide/comptime.md

```diff
@@ -11,7 +11,7 @@ kernelspec:
 When writing Guppy code, it's important to understand when different parts of our program are executed. We distinguish between two key stages:
 
 - **Compile-time** refers to everything that happens on our local machine before submitting the program to an emulator or quantum device.
-  This includes the work the Guppy compiler does when calling ``guppy.compile()``, but also any other Python code that is run along-side.
+  This includes the work the Guppy compiler does when calling ``guppy.compile_function()``, but also any other Python code that is run along-side.
```

Comment:
Maybe readers could be confused why this function is only used in these examples? Might be adding something about how `comptime` things are only evaluated when calling compile rather than check and how we want to quickly check things that take arguments without necessarily compiling something that can run

I know @mark-koch wants to run comptime things on `check` but that is likely not a quick change

Reply:
Going to merge this and add the note in a follow up.